### ISSUE TITLE
Support sprintf for ServiceKey?

### ISSUE THREAD

**Issue by cschotke** [logstash-plugins/logstash-output-pagerduty#2]
Is there any reason why the Service Key field does not support sprintf but some of the other fields do? We would like to be able to pull the service key value from a field in the event instead of having to hardcode it in the config. I tested changing 

```
pd_event[:service_key] = "#{@service_key}"
```

to 

```
pd_event[:service_key] = event.sprintf(@service_key)
```

and everything seemed to work.

**Comment by untergeek** [comment]
This is an interesting point.  We should investigate allowing it to be either hard-coded, or a named field.

**Comment by cgspohn** [comment]
I am interested in the same enhancement, since we would like to define the value before hitting the output section.  Any chance this will be implemented?

**Comment by magnusbaeck** [comment]
Same request was made in https://discuss.elastic.co/t/logstash-fields-not-resolving-pagerduty-plugin/81046 and not only for `service_key` but also `event_type`. The limitation of the latter is easier to work around since it accepts a discrete set of values so you could declare multiple pagerduty outputs with different `event_type` values and use conditionals to pick one of them.

**Comment by adawalli** [comment]
https://github.com/logstash-plugins/logstash-output-pagerduty/pull/16 intends to fix this.

**Comment by adawalli** [comment]
Note, this PR doesn't attempt to fix the "event" situation. This is simply a fix for the **Service Key**

**Comment by colinsurprenant** [comment]
Fixed in #16 - closing.
